src/univariate/continuous/chernoff.jl

```diff
@@ -0,0 +1,318 @@
+    else
+        return sum([exp(cuberoottwo*a*y) for a in az]) * 2 * sqrttwopi * exp(-y*y*y/6) - cnsty
+end
+
+function g(x::Real)
+    function gone(y::Real)
+        return p(y) * exp(-0.5*y*(2*x+y)*(2*x+y))
+    end
+    function gtwo(y::Real)
+        z = 2*x+y*y
+        return (z*y*y + 0.5 * z*z) * exp(-0.5*y*y*z*z)
+    end
+    if (x <= -1.0)
+        return cuberoottwo*cuberoottwo * exp(2*x*x*x/3.0) * sum([exp(-cuberoottwo*airyai_roots[k]*x) / airyai_prime[k] for k=1:length(airyai_roots)])
+    else
+        return 2*x - (quadgk(gone, 0.0, Inf)[1] - 4*quadgk(gtwo, 0.0, Inf)[1]) / sqrttwopi   # should perhaps combine integrals
+    end
+end
+
+_pdf(x::Real) = g(x)*g(-x)*0.5
+_cdf(x::Real) = (x < 0.0) ? _cdfbar(-x) : 0.5 + quadgk(_pdf,0.0,x)[1] 
+_cdfbar(x::Real) = (x < 0.0) ? _cdf(x) : quadgk(_pdf, x, Inf)[1]
+
+
+
+pdf(d::Chernoff, x::Real) = _pdf(x)
+logpdf(d::Chernoff, x::Real) = log(g(x))+log(g(-x))+log(0.5)
+cdf(d::Chernoff, x::Real) = _cdf(x)
+
+function quantile(d::Chernoff, tau::Real)
+    # some commonly used quantiles were precomputed
+    precomputedquants=[
+        0.0 -Inf;
+        0.01 -1.171534341573129;
+        0.025 -0.9981810946684274;
+        0.05 -0.8450811886357725;
+        0.1 -0.6642351964332931;
+        0.2 -0.43982766604886553;
+        0.25 -0.353308035220429;
+        0.3 -0.2751512847290148; 
+        0.4 -0.13319637678583637;
+        0.5 0.0;
+        0.6 0.13319637678583637;
+        0.7 0.2751512847290147;
+        0.75 0.353308035220429;
+        0.8 0.4398276660488655;
+        0.9 0.6642351964332931;
+        0.95 0.8450811886357724;
+        0.975 0.9981810946684272;
+        0.99 1.17153434157313;
+        1.0 Inf
+        ]
+    (0.0 <= tau && tau <= 1.0) || throw(DomainError(tau, "illegal value of tau"))
+    present = searchsortedfirst(precomputedquants[:, 1], tau)
+    if present <= size(precomputedquants, 1)
+        if tau == precomputedquants[present, 1]
+            return precomputedquants[present, 2] 
+        end
+    end
+
+    dnorm = Normal(0.0, 1.0)
+    if tau < 0.001 
+        return -newton(x -> tau - _cdfbar(x), _pdf, quantile(dnorm, 1.0 - tau)*0.52) 
+
+    end
+    if tau > 0.999 
+        return newton(x -> 1.0 - tau - _cdfbar(x), _pdf, quantile(dnorm, tau)*0.52) 
+    end
+    return newton(x -> _cdf(x) - tau, _pdf, quantile(dnorm, tau)*0.52)   # should consider replacing x-> construct for speed
+end
+
+minimum(d::Chernoff) = -Inf
+maximum(d::Chernoff) = Inf
+insupport(d::Chernoff, x::Real) = isnan(x) ? false : true
+
+mean(d::Chernoff) = 0.0
+var(d::Chernoff) = 0.26355964132470455
+modes(d::Chernoff) = [0.0]
+mode(d::Chernoff) = 0.0
+skewness(d::Chernoff) = 0.0
+kurtosis(d::Chernoff) = -0.16172525511461888
+kurtosis(d::Chernoff, excess::Bool) = kurtosis(d) + (excess ? 0.0 : 3.0) 
+entropy(d::Chernoff) = -0.7515605300273104
+
+### Random number generation
+slowrand(d::Chernoff) = quantile(d, rand(GLOBAL_RNG))
```

Comment:
There should be a docstring for when using `slowrand` is interesting, if it's always slower we can remove it, there is a default for uniform sampling + quantile, no need to define it for `Chernoff`

Reply:
As of now, I can't seem to come up with a use case for this, should I `remove` it?

Reply:
I think this can be removed